### ISSUE TITLE
Vi mode configuration in prompt causes bindkey conflict

### ISSUE THREAD

**Issue by CadaverLab** [marlonrichert/zsh-autocomplete#23]
I use vi mode and have my prompt set up for it accordingly. Here's the relevant code:

```
function zle-keymap-select {
  if [[ ${KEYMAP} == vicmd ]] ||
     [[ $1 = 'block' ]]; then
    echo -ne '\e[2 q'
  elif [[ ${KEYMAP} == main ]] ||
       [[ ${KEYMAP} == viins ]] ||
       [[ ${KEYMAP} = '' ]] ||
       [[ $1 = 'beam' ]]; then
    echo -ne '\e[6 q'
  fi
}
zle -N zle-keymap-select
zle-line-init() {
    echo -ne "\e[6 q"
}
zle -N zle-line-init
echo -ne '\e[5 q' # Use beam shape cursor on startup.
preexec() { echo -ne '\e[6 q' ;} # Use beam shape cursor for each new prompt.
```

What this does is change the cursor from beam to block depending on the mode (beam for insert, block for normal)

If i run `bindkey "^I"` with that block present, i get `"^I" fzf-completion` as a result. If i remove it, `bindkey "^I"` prints `"^I" complete-word`.

My steps for tracking this down were creating a minimal .zshrc that only sourced fzf and the plugin, and then re-adding everything back little by little.

**Comment by marlonrichert** [comment]
Instead of using `zle -N zle-keymap-select` and `zle -N zle-line-init`, can you try using [`add-zle-hook-widget`](http://zsh.sourceforge.net/Doc/Release/User-Contributions.html#Manipulating-Hook-Functions) instead?

**Comment by marlonrichert** [comment]
I tested your case again with the latest code and I can no longer reproduce it. It seems to have been fixed by 16068f35d1a2be01f396b57544f3cf9b17285c82. Please update and try again. 🙂

**Comment by CadaverLab** [comment]
I just pulled and tested and it works, thank you.

I'm gonna try using the way you suggested but i need to do some reading up on it first
